catalogs/catalog-kafka/src/test/java/com/datastrato/gravitino/catalog/kafka/integration/CatalogKafkaIT.java

```diff
@@ -0,0 +1,370 @@
+    Assertions.assertEquals(
+        Integer.parseInt(createdTopic.properties().get(PARTITION_COUNT)),
+        topicDesc.partitions().size());
+    Assertions.assertEquals(
+        Integer.parseInt(createdTopic.properties().get(REPLICATION_FACTOR)),
+        topicDesc.partitions().get(0).replicas().size());
+
+    // get properties from Kafka directly
+    ConfigResource configResource =
+        new ConfigResource(ConfigResource.Type.TOPIC, createdTopic.name());
+    Config topicConfigs =
+        adminClient
+            .describeConfigs(Collections.singleton(configResource))
+            .all()
+            .get()
+            .get(configResource);
+    topicConfigs
+        .entries()
+        .forEach(
+            entry ->
+                Assertions.assertEquals(
+                    entry.value(), createdTopic.properties().get(entry.name())));
+  }
+
+  private void checkTopicReadWrite(String topicName) {
+    String randomMessage = GravitinoITUtils.genRandomName("message");
+    int partition = -1;
+    long offset = -1;
+    // write message to topic
+    try (KafkaProducer<String, String> producer =
+        new KafkaProducer<>(
+            ImmutableMap.of(
+                ProducerConfig.BOOTSTRAP_SERVERS_CONFIG,
+                KAFKA_BOOTSTRAP_SERVERS,
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                StringSerializer.class.getName(),
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                StringSerializer.class.getName()))) {
+      ProducerRecord<String, String> record =
+          new ProducerRecord<>(topicName, randomMessage, randomMessage);
+      RecordMetadata metadata = producer.send(record).get();
+      Assertions.assertEquals(topicName, metadata.topic());
+      offset = metadata.offset();
+      partition = metadata.partition();
+      Assertions.assertTrue(offset >= 0);
+      Assertions.assertTrue(partition >= 0);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to write message to Kafka", e);
+    }
+
+    // read message from topic
+    try (KafkaConsumer<String, String> consumer =
+        new KafkaConsumer<>(
+            ImmutableMap.of(
+                ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+                KAFKA_BOOTSTRAP_SERVERS,
+                ConsumerConfig.GROUP_ID_CONFIG,
+                topicName,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName(),
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName(),
+                ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+                "false"))) {
+      TopicPartition topicPartition = new TopicPartition(topicName, partition);
+      consumer.assign(ImmutableList.of(topicPartition));
+      consumer.seek(topicPartition, offset);
+      ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(3));
```

Comment:
3 seconds is too short; the test may fail if there are insufficient resources.

Reply:
How long do you suggest setting it for?

Reply:
30s